src/backend/access/external/hd_work_mgr.c

```diff
@@ -783,6 +786,10 @@ make_allocation_output_string(List *segment_fragments)
 		appendStringInfoChar(&fragment_str, SEGWORK_IN_PAIR_DELIM);
 		if (frag->fragment_md)
 			appendStringInfo(&fragment_str, "%s", frag->fragment_md);
+		/* Adding dfs_address from pg_filespace entry required for HAWQ-462 */
+		appendStringInfoChar(&fragment_str, SEGWORK_IN_PAIR_DELIM);
+		if (dfs_address)
```

Comment:
Yes, this is the first cut solution, the optimal solution is to pass it as part of the segment level data instead of fragment. Wasn't quite able to figure out how to add additional metadata with the segment data. Wanted to first certify against secure cluster before investigating addition into segment data.